### ISSUE TITLE
sui_getCheckpoint returns an error for some checkpoints

### ISSUE THREAD

**Issue by jim423** [MystenLabs/sui#14417]
## Steps to Reproduce Issue

I tried to get the timestamp of a given checkpoint via sui_getCheckpoint API. However, the API does not return any checkpoint information, instead, it returns an error if there is a digest not found.

By calling sui_getCheckpoint for the checkpoint "12299891"
```
curl --location --request POST 'https://fullnode.mainnet.sui.io' \
--header 'Content-Type: application/json' \
--data-raw '{
    "jsonrpc": "2.0",
    "id": "1",
    "method": "sui_getCheckpoint",
    "params": [
        "12299891"
    ]
}'
```

I expected the result should return all the checkpoint info without an error.
## Expected Result
```
{
    "jsonrpc": "2.0",
    "result": {
        "epoch": "147",
        "sequenceNumber": "12299891",
        "digest": "3JcdR5EN9GM48F26CyCZQFNphsj56HkJDpZcQyiW2y84",
        "networkTotalTransactions": "787256771",
        "previousDigest": "8gN8ZxdgebY29riioMDorhQb3sNUFAdd7AKvGdJ4gasF",
        "epochRollingGasCostSummary": {
            "computationCost": "185598939592",
            "storageCost": "2531800182800",
            "storageRebate": "1985615373852",
            "nonRefundableStorageFee": "20056720948"
        },
        "timestampMs": "1694031746475",
        "transactions": [
            "DZ19BMiaWNjTVJoGqvWTo6aBXH4qZnWuVUfJJqNLuLAj",
            "6ivfh8orB4R67gug9wn8igwniKBpJ4SKoLwURzBiz4S",
            "4T9a5vV6bErhV6RnnP6QnYRGJJKqankJEuEQsVFKLrLm",
            "6D5krHShCYwvXWesWjtkpG4WMLfp7ZMEsC3oL7hF9gmN",
            "73FgeEaSE27dHEcS2Z5WprsSJ9WsCtHuRCmgHhvgPXyN",
            "BsHTAbn2bwLfvQAYcomJYMQNW78N4iC5k9ZGkhzvVXjA",
            "Ao149PQAgnBtjw6zMbatfroMWrVtnRPgQxZNVaZQ4V1k"
        ],
        "checkpointCommitments": [],
        "validatorSignature": "ihabn7i/nHo2JQMSdv/vBJZK6XEg1Vxck8US2B1NcjloMXEAshP3sazNGuwGuAw/"
    },
    "id": "1"
}
```

 But it returns an error when there is a digest not found of the given checkpoint.
## Actual Result
```
{
    "jsonrpc": "2.0",
    "error": {
        "code": -32602,
        "message": "Verified checkpoint not found for digest: CheckpointContentsDigest(7UoFkreuL7aEvwDCpNgaZgYVDrosFkhr5vZ8ZAmtyEyC)"
    },
    "id": "1"
}
```

If that is the behaviour of sui_getCheckpoint, what is the workaround to get the timestamp of a specific checkpoint?

**Comment by stefan-mysten** [comment]
What have you described should work. It might an issue on our side, I flagged it internally to the team and I will update here asap. Thanks for flagging this!

**Comment by stefan-mysten** [comment]
This should have been fixed with the last `mainnet` release. @jim423 could you please confirm that it's working now?

**Comment by jim423** [comment]
@stefan-mysten I have confirmed that the issue has been addressed. Thank you so much for the update and efforts.

**Comment by jim423** [comment]
@stefan-mysten  By the way, will this fix be available on the latest release version?

**Comment by jim423** [comment]
@stefan-mysten Cuz I did not see the issue number listed in the release log, I was wondering if the fix was included in the Mainnet v1.12.2 release version.

**Comment by stefan-mysten** [comment]
Which issue number are you referring to, this issue's number #14417? I am certain that the fix is included in the `v1.12.2` release too.

**Comment by jim423** [comment]
Thanks for the confirmation!
